### DISCUSSION TITLE
Self-hosting

### DISCUSSION THREAD

**Discussion by notpushkin** [govuk-forms/forms#106] [Q&A] [answered]
Hey, first of all thanks for your wonderful project!

I was wondering if it is possible to self-host GOV.UK Forms to try it out without being a GDS employee. I've built [Docker images](https://github.com/notpushkin?tab=packages&tab=packages&q=govuk-forms-) for each application and wrote a simple [signon service stub](https://gitlab.com/notpushkin/stupidauth), then I ran [everything together](https://gist.github.com/notpushkin/f174b52efc6285e6be30fb66bb48153f) as a Docker stack. Currently, I'm getting a generic error after I try to sign in and nothing else happens. Nothing interesting [in the logs](https://gist.github.com/notpushkin/7616f2fa9662bfc5580cbaa1d3d9bbc5), too.

I understand this is likely not a goal you had in mind when releasing this, but if you could give a hint on how to proceed I would be extremely grateful!

**Comment by chao-xian** [comment, marked as answer]
Hi! Thanks for your feedback on the project. We don't support self-hosting outside of our own usage unfortunately. We are more "source available" really (hence the MIT licence). Please do let us know if you manage to overcome your issue though. 

If you’re working with another UK Government department and are interested in GOV.UK Forms, you can contact our engagement team via the Register your interest button or contact on https://www.forms.service.gov.uk/
